src/cmake/compiler.cmake

```diff
@@ -283,6 +291,10 @@ if (SANITIZE AND (CMAKE_COMPILER_IS_GNUCC OR CMAKE_COMPILER_IS_CLANG))
         set (SANITIZE_LIBRARIES "asan;pthread")
         # set (SANITIZE_LIBRARIES "asan" "ubsan")
     endif()
+    if (CMAKE_COMPILER_IS_GNUCC)
+        # turn on glibcxx extra annotations to find vector writes past end
+        add_definitions ("-D_GLIBCXX_SANITIZE_VECTOR=1")
```

Comment:
Same with this one, it works for clang too, its a `libstdc++` feature, not a gcc one.